xt/check-signatures.t

```diff
@@ -0,0 +1,146 @@
+use Test;
+use Telemetry; # so we can check it
+grammar TypeDocumentation {...}
+
+=begin SYNOPSIS
+Check the signatures of documented methods against the Rakudo source code.
+
+For each method documented in a .pod6 file in doc/doc/Type/, this test compares
+the documented signature against the signature in the Rakudo source code.
+Because we are documenting I<Raku> and not I<Rakudo>, detecting a difference in
+the signature does not automatically cause the test to fail: some differences in
+signature (such as different names for positional parameters) represent
+different implementation choices rather than an error in the docs.
+
+To ensure that implementation details don't cause failing tests, we check only
+for certain discrepancies that are guaranteed to indicate a
+substantive/non-implementation detail mismatch between Rakudo and the docs.
+
+Currently, we only test for one category of discrepancies:
+
+=item Methods that are defined with a specific invocant in Rakudo but not in the
+docs (this helpfully also catches the situation where a documentation signature
+was I<intended> to have an invocant, but where someone forgot to end the
+invocant with a C<:>)
+
+In the future, this test could be expanded to also check for:
+
+=item arity mismatches
+=item different names for named parameters
```

Comment:
This can happen. I don't think this should be really checked. In some cases, names can be changed for clarification.

Reply:
Thanks.  I agree that names shouldn't be checked for _positional_ parameters (in fact, I mentioned that as the example of something _not_ to check).  But if we give the wrong name for a _named_ parameter, then a user following the docs wouldn't be able to use the method (e.g., if they passed in `:key` to `grep` instead of `:k`, the method won't work)

Reply:
Ah, sorry, yes, of course.